### PR TITLE
Add device permissions

### PR DESCRIPTION
**Pull request by slukic0**
- Add deviceAdmins table to store device admins and authorized users
- Add device managment endpoints for registering devices, adding users, and removing users
- Check user permissions when getting sensor data

### CODE REVIEW

File: air-quality-sst/packages/functions/src/devices.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        if (session.type !== 'user') {
+            return createJsonMessage(401, 'Unauthorized')
+        }
+
+        const data = JSON.parse(event?.body || '')
+        if (!data?.deviceId) {
+            createJsonMessage(400, 'deviceId is required')
+        }
+
+        // Register device ro user if the device is not registered
```
Comment (FLukic0) [inline air-quality-sst/packages/functions/src/devices.ts]: to

File: air-quality-sst/packages/functions/src/devices.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+        // Create transaction
+        const params = {
+            TransactItems: [{ Put: PutDeviceAdmin }, { Update: UpdateUsers }],
+        }
+
+        // Return result
+        try {
+            const results = await dynamoDb.transactWrite(params).promise()
+            console.log(results)
```
Comment (FLukic0) [inline air-quality-sst/packages/functions/src/devices.ts]: maybe remove console.log once done

**Review by FLukic0 — APPROVED** [review]

File: air-quality-sst/packages/functions/src/devices.ts (hunk 2 above)
Comment (slukic0) [reply air-quality-sst/packages/functions/src/devices.ts]: no

Comment (FLukic0) [reply air-quality-sst/packages/functions/src/devices.ts]: what you mean nuh uh

**Pull request merged**